openedx/core/djangoapps/content/search/documents.py

```diff
@@ -230,6 +230,7 @@ def searchable_doc_for_course_block(block, metadata: bool = True, tags: bool = T
     doc = {
         Fields.id: meili_id_from_opaque_key(block.usage_key),
         Fields.type: DocType.course_block,
+        Fields.tags: {},
```

Comment:
This line will clear the tags if we update an XBlock. Maybe the change in `_tags_for_content_object()` already fixed it? (I removed this line while testing and it seems to work)

```suggestion
```

Reply:
You're right, I added this mainly so all the docs have the same format that has (containing `tags` field, whether or not there are tags), however this caused the issue you mentioned, so I removed it.